providers/org/iacr/harvester.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+from lxml import etree
+from share import Harvester
+
+logger = logging.getLogger(__name__)
+
+
+class IACRHarvester(Harvester):
+
+    namespaces = {
+        'ns0': 'http://www.w3.org/2005/Atom'
+    }
+
+    url = 'https://eprint.iacr.org/rss/rss.xml'
+
+    def do_harvest(self, start_date, end_date):
+        # IACR has previous-search, you can only go from some past day to today
+        return self.fetch_records(self.url, start_date)
+
+    def fetch_records(self, url, start_date):
+        return self.fetch_page(url)
+
+    def fetch_page(self, url):
+        logger.info('Making request to {}'.format(url))
+
+        resp = self.requests.get(url, verify=False)
+        parsed = etree.fromstring(resp.content)
+        total_records = int(parsed.xpath("//ttl")[0].text)
+
+        records = parsed.xpath('//item', namespaces=self.namespaces)
+        logger.info('Found {} records of {}.'.format(len(records), total_records))
+        i = 0
+        for record in records:
+            doc_id = record.xpath('//guid', namespaces=self.namespaces)[i].text
```

Comment:
Looks like an oddity of how xpath works. `//<tag>` looks at the entire document. You can limit the scope with `./<tag>`

```python
ipdb> record.xpath('//guid')
[<Element guid at 0x10840a488>, <Element guid at 0x10840a188>, <Element guid at 0x10840a3c8>, <Element guid at 0x10840acc8>, <Element guid at 0x10840a108>, <Element guid at 0x10840a048>, <Element guid at 0x10840ab08>, <Element guid at 0x10840ac48>, <Element guid at 0x10840ab88>, <Element guid at 0x10840a708>, <Element guid at 0x10840afc8>, <Element guid at 0x10840abc8>, <Element guid at 0x10840a808>, <Element guid at 0x10840a088>, <Element guid at 0x10840a648>, <Element guid at 0x10840ab48>, <Element guid at 0x10840a948>, <Element guid at 0x10840a2c8>, <Element guid at 0x10840a848>, <Element guid at 0x10840ac08>, <Element guid at 0x10840f888>, <Element guid at 0x10840f048>, <Element guid at 0x10840f088>, <Element guid at 0x10840f6c8>, <Element guid at 0x10840f288>, <Element guid at 0x10840f508>, <Element guid at 0x10840f3c8>, <Element guid at 0x10840f148>, <Element guid at 0x10840f108>, <Element guid at 0x10840f488>, <Element guid at 0x10840f0c8>, <Element guid at 0x10840f388>, <Element guid at 0x10840f2c8>, <Element guid at 0x10840f188>, <Element guid at 0x10840f248>, <Element guid at 0x10840f4c8>, <Element guid at 0x10840f308>, <Element guid at 0x10840f648>, <Element guid at 0x10840f1c8>, <Element guid at 0x10840f548>, <Element guid at 0x10840f448>, <Element guid at 0x10840f408>, <Element guid at 0x10840fa48>, <Element guid at 0x10840f9c8>, <Element guid at 0x10840f948>, <Element guid at 0x10840f708>, <Element guid at 0x10840f588>, <Element guid at 0x10840f208>, <Element guid at 0x10840fc88>, <Element guid at 0x10840f848>, <Element guid at 0x10840fd88>, <Element guid at 0x10840f688>, <Element guid at 0x10840fa08>, <Element guid at 0x10840f748>, <Element guid at 0x10840fe08>, <Element guid at 0x10840f348>, <Element guid at 0x10840f908>, <Element guid at 0x10840f608>, <Element guid at 0x10840fa88>, <Element guid at 0x10840fb08>, <Element guid at 0x10840f788>, <Element guid at 0x10840f988>, <Element guid at 0x10840f8c8>, <Element guid at 0x10840fec8>, <Element guid at 0x10840f5c8>, <Element guid at 0x10840fc48>, <Element guid at 0x10840ffc8>, <Element guid at 0x10840ff48>, <Element guid at 0x10840fac8>, <Element guid at 0x10840f7c8>, <Element guid at 0x10840fd08>, <Element guid at 0x10840fe48>, <Element guid at 0x10840fb48>, <Element guid at 0x10840fbc8>, <Element guid at 0x10840fd48>, <Element guid at 0x10840fe88>, <Element guid at 0x10840fdc8>, <Element guid at 0x10840fc08>, <Element guid at 0x10840fcc8>, <Element guid at 0x10840f808>, <Element guid at 0x1083f5848>, <Element guid at 0x1083f56c8>, <Element guid at 0x1083f54c8>, <Element guid at 0x1083f5348>, <Element guid at 0x1083f5588>, <Element guid at 0x1083f5148>]

ipdb> record.xpath('./guid')
[<Element guid at 0x10840a488>]
```